libbeat/outputs/elasticsearch/config_presets.go

```diff
@@ -0,0 +1,130 @@
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package elasticsearch
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/elastic/elastic-agent-libs/config"
+)
+
+const (
+	presetNone       = ""
+	presetCustom     = "custom"
+	presetBalanced   = "balanced"
+	presetThroughput = "throughput"
+	presetScale      = "scale"
+	presetLatency    = "latency"
+)
+
+var presetConfigs = map[string]*config.C{
+	presetNone:   config.MustNewConfigFrom(map[string]interface{}{}),
+	presetCustom: config.MustNewConfigFrom(map[string]interface{}{}),
+	presetBalanced: config.MustNewConfigFrom(map[string]interface{}{
+		"bulk_max_size":              1600,
+		"worker":                     1,
+		"queue.mem.events":           3200,
+		"queue.mem.flush.min_events": 1600,
+		"queue.mem.flush.timeout":    10 * time.Second,
+		"compression_level":          1,
+		"idle_connection_timeout":    3 * time.Second,
+	}),
+	presetThroughput: config.MustNewConfigFrom(map[string]interface{}{
+		"bulk_max_size":              1600,
+		"worker":                     4,
+		"queue.mem.events":           12800,
+		"queue.mem.flush.min_events": 1600,
+		"queue.mem.flush.timeout":    5 * time.Second,
+		"compression_level":          1,
+		"idle_connection_timeout":    15 * time.Second,
+	}),
+	presetScale: config.MustNewConfigFrom(map[string]interface{}{
+		"bulk_max_size":              1600,
+		"worker":                     1,
+		"queue.mem.events":           3200,
+		"queue.mem.flush.min_events": 1600,
+		"queue.mem.flush.timeout":    20 * time.Second,
+		"compression_level":          1,
+		"idle_connection_timeout":    1 * time.Second,
+	}),
+	presetLatency: config.MustNewConfigFrom(map[string]interface{}{
+		"bulk_max_size":              50,
+		"worker":                     1,
+		"queue.mem.events":           4100,
+		"queue.mem.flush.min_events": 2050,
+		"queue.mem.flush.timeout":    1 * time.Second,
+		"compression_level":          1,
+		"idle_connection_timeout":    60 * time.Second,
+	}),
+}
+
+// Given a user config, check its preset field and apply any corresponding
+// config overrides.
+// Returns a list of the user fields that were overwritten.
+func applyPreset(preset string, userConfig *config.C) ([]string, error) {
+	presetConfig := presetConfigs[preset]
```

Comment:
Might want to do `presetConfig, ok :=` to check if the preset exists?

Reply:
That's what the `presetConfig == nil` check on the next line is for, the `ok` check wouldn't catch anything that `nil` doesn't already